### ISSUE TITLE
Unable to reproduce the performance in paper

### ISSUE THREAD

**Issue by leeerchan** [georgeNakayama/AIpparel-Code#9]
I ran `bash eval_scripts/run_eval_image.sh` to evaluate image-2-pattern performance. Since I have downloaded GarmentCodeDataV2 (not V1), I removed datapoints in `garmentcodedata_datasplit.json` that are not included in the V2 dataset, reducing the size of the validation set from 6,381 to 5,891.  The output metrics were

![Image](https://github.com/user-attachments/assets/45987e24-8a62-4a3d-9190-e30312b44578)

However, the results in paper are

![Image](https://github.com/user-attachments/assets/44e9dd32-e36b-49cc-8523-0e48b95472aa)

I've created the shifted specification files, fix the bug in https://github.com/georgeNakayama/AIpparel-Code/issues/8.

Could you provide suggestions or guidance on whether I might have made an error in any step?

**Comment by georgeNakayama** [comment]
Hi, 

This is weird.. The evaluation does have some stochasticity where it picks either a front or back image for each data sample. However, the numbers do look a bit worse than what it supposed to be. 

Can you send me your split json and I can take a look on my end? 

Thanks!

**Comment by leeerchan** [comment]
Thanks for your reply! This is the split file I used: 

[garmentcodedata_datasplit_v2.json](https://github.com/user-attachments/files/20226827/garmentcodedata_datasplit_v2.json)

**Comment by georgeNakayama** [comment]
Hi, 

somehow I similar numbers to the paper when I ran it, using your split. 

![Image](https://github.com/user-attachments/assets/bf64d3f2-f0a5-4c6d-a933-f566635e20d6)

It might be due to the shifting being incorrect, but I'm not sure. I'll just upload the shifted pattern jsons from my end to Huggingface, and can you try evaluating using that?

**Comment by georgeNakayama** [comment]
Hi, 

The shifted json files are uploaded to https://huggingface.co/georgeNakayama/AIpparel/blob/main/garment_specifications_archive.tar.gz

Please download and compress them under the garmentcodedata directory. They will write relative path from there to each pattern folder. 

Please let me know if this works!

**Comment by phphuc612** [comment]
Hi @georgeNakayama,
I have reproduced the image-text tables with the input the render front view (using V1 image).
I run the test set provided in your [assets/data_configs/garmentcodedata_datasplit_v2.json](https://github.com/georgeNakayama/AIpparel-Code/blob/master/assets/data_configs/garmentcodedata_datasplit_v2.json).
Could you please take a look if this is expected performance?
There is considerable decrease on stitching accuracy and correct edge accuracy while other metrics are better than reported.

I truly appreciate your help.

<img width="1502" height="41" alt="Image" src="https://github.com/user-attachments/assets/6ca52143-f73e-4c49-8423-3d18e86f38fc" />
Comparing side-by-side:

<img width="880" height="53" alt="Image" src="https://github.com/user-attachments/assets/da9aa3c3-876d-449c-8ddc-b974973b294b" />

**Comment by georgeNakayama** [comment]
Hi, how long have you trained the model for? The evaluation script currently will take either the front or back image of each evaluation data sample and run the garment reconstruction. So depending on which image it took the metric will be different. 

Also would be beneficial to take a look at the reconstructed garments, since under-converged model would produce less regular panel shapes.

**Comment by phphuc612** [comment]
Hi @georgeNakayama, I'm sorry for the confusion here. I use your checkpoint to run this metric. 
The difference is that I force all samples to take the front view and I hash name of sample `rand_xxxx` and set it as seed at the beginning of `__getitem__` to make prompt sampling deterministic.

Front view:
```python
def _parepare_image(self, image_paths):
        """Fetch the image for the given index"""
        if self.is_training:
            image_path = random.choice(image_paths)
        else:  # for evaluation, use front image
            if "_render_front" in image_paths[0]:
                image_path = image_paths[0]
            else:
                image_path = image_paths[-1]
        ...
```

Set seed:
```python
    def __getitem__(self, idx):
        """Called when indexing: read the corresponding data.
        Does not support list indexing"""
        if torch.is_tensor(idx):  # allow indexing by tensors
            idx = idx.tolist()

        datapoint_name = self.datapoints_names[idx]
        data_name = datapoint_name.split("/")[-1]
        # deterministic sampling
        # Hash the name to a reproducible integer
        hash_bytes = hashlib.sha256(data_name.encode("utf-8")).digest()
        # Convert first 4 bytes to an integer (fits into 32-bit)
        seed = int.from_bytes(hash_bytes[:4], "big")

        # Set both numpy and random seed
        np.random.seed(seed)
        random.seed(seed)
        ....
```
